samples/QuickstartWeatherServer/Tools/WeatherTools.cs

```diff
@@ -42,8 +42,11 @@ public static async Task<string> GetForecast(
         [Description("Longitude of the location.")] double longitude)
     {
         using var jsonDocument = await client.ReadJsonDocumentAsync($"/points/{latitude},{longitude}");
-        var jsonElement = jsonDocument.RootElement;
-        var periods = jsonElement.GetProperty("properties").GetProperty("periods").EnumerateArray();
+        var properties = jsonDocument.RootElement.GetProperty("properties");
+        var forecastUrl = properties.GetProperty("forecast").GetString();
+
+        using var forecastDocument = await client.ReadJsonDocumentAsync(forecastUrl ?? $"/{properties.GetProperty("gridId")}/{properties.GetProperty("gridX")},{properties.GetProperty("gridY")}/forecast");
```

Comment:
```suggestion
        var forecastUrl = jsonDocument.RootElement.GetProperty("properties").GetProperty("forecast").GetString()
            ?? throw new Exception($"No forecast URL provided by {client.BaseAddress}points/{latitude},{longitude}");

        using var forecastDocument = await client.ReadJsonDocumentAsync(forecastUr);
```

Reply:
Agree @halter73. I've applied your suggestion so we can merge it out and let more improvements to come with the most basic scenario working. 

Thanks!